internal/transport/http2_client.go

```diff
@@ -778,6 +770,11 @@ func (t *http2Client) Close() error {
 	t.state = closing
 	streams := t.activeStreams
 	t.activeStreams = nil
+	if t.kpDormancyCond != nil {
+		// If the keepalive goroutine is blocked on this condition variable, we
+		// should unblock it so that the goroutine eventually exits.
```

Comment:
There is a small race here.  If the `t.kp.Time` timer just fired, we may `Signal()` here, but before we reach the `Wait()` in the keepalive goroutine.  This would leave it orphaned forever.

One solution would be to make `shouldKeepaliveBeActive()` always return `true` on a transport that is shut down - and ensure this happens before we `Signal()` so that the keepalive goroutine never calls `Wait()` after this.